adb-shell-pm-list-packages: add page

UsmanAhmadSaeed opened the pull request:
adb-pm-list-packages: add file

<!--
Thank you for contributing!
Please fill in the following checklist, removing items that do not apply.
See also https://github.com/tldr-pages/tldr/blob/main/CONTRIBUTING.md
-->

- [X] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [X] The page description(s) have links to documentation or a homepage.
- [X] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [X] The page(s) follow the [style guide](/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md).
- [X] The PR contains at most 5 new pages.
- [X] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):**
Addresses adb>pm>list>packages under #13997

Managor commented:
Start by first creating a page for the base `adb shell pm` before going any deeper into the subcommands. I'll close this for now. We'll reopen this once you've built up the foundations.

Managor commented on pages/common/adb-pm-list-packages.md:
```diff
@@ -0,0 +1,33 @@
+# pm list packages
+
+> List installed, known, or filtered packages on an Android device via Package Manager.
+> Used with `adb shell`. Various flags allow filtering and extra package information.
```
```suggestion
> List installed, known, or filtered packages on an Android device.
```
I didn't notice previously that this is redundant

UsmanAhmadSaeed commented:
Updated. Also added adb shell to file name and header.
Please also label approved PRs (this, #18792, #18795) with hacktoberfest-accepted tag for Hacktoberfest 2025.

nelsonfigueroa reviewed (approved):
Thank you for your contribution!
